storage/aptosdb/src/lib.rs

```diff
@@ -1295,6 +1297,11 @@ impl DbWriter for AptosDB {
         save_state_snapshots: bool,
     ) -> Result<()> {
         gauged_api("save_transactions", || {
+            let _lock = self
+                .ledger_commit_lock
+                .try_lock()
+                .expect("Concurrent committing detected.");
```

Comment:
why panic? just wait and lock? this seems more like a debug_assert

Reply:
We don't support committing an old commit anyway, I'd make a larger noise to be noticed.

Reply:
May be add a comment here to make it clearer?